src/accelerate/tracking.py

```diff
@@ -593,7 +593,7 @@ def filter_trackers(
                             if getattr(tracker_init, "requires_logging_directory"):
                                 if logging_dir is None:
                                     raise ValueError(
-                                        f"Logging with `{str(log_type)}` requires a `logging_dir` to be passed in."
+                                        f"Logging with `{log_type}` requires a `logging_dir` to be passed in."
```

Comment:
This is not equivalent, so this change should be reverted (f-strings use the repr, `str` does not).

Reply:
For the record I quickly tested it and it seems that it actually uses `__str__`, not `__rep__`. 

```py
class A:
    def __str__(self):
        return "Use str"

    def __repr__(self):
        return "Use repr"

a = A()
print(a)
print("%s" % a)
print(f"{a}")
print(f"{str(a)}")
print("{}".format(a))
```

Output (both in python 3.7 and 3.10.
```
Use str
Use str
Use str
Use str
Use str
```

Reply:
Oh! My bad then.

Reply:
It is actually a TIL for me, didn't know what to expect :)